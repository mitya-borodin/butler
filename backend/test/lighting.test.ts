import fetch from "node-fetch";

const BASE_URL = "http://localhost:5000";

afterAll(async () => {
  await fetch(`${BASE_URL}/purge-test-database`, { method: "POST" });
});

test("Create lightning device", async () => {
  const response = await fetch(`${BASE_URL}/create-lighting-device`, {
    method: "PUT",
    headers: {
      "Content-type": "application/json",
    },
    body: JSON.stringify([
      {
        name: "lightning-device-1",
        brand: "FREON",
        power: "24",
        lumens: "4000",
        lightTemperatureKelvin: 0,
        resourceMs: 0,
        price: "600",
        currency: "RUB",
        sellersWebsite: "http://feron.ru",
        images: ["image_1", "image_2", "image_3"],
        placeOfInstallation: "Кухня",
        state: "OFF",
      },
      {
        name: "lightning-device-2",
        brand: "FREON",
        power: "24",
        lumens: "4000",
        lightTemperatureKelvin: 0,
        resourceMs: 0,
        price: "600",
        currency: "RUB",
        sellersWebsite: "http://feron.ru",
        images: ["image_1", "image_2", "image_3"],
        placeOfInstallation: "Кухня",
        state: "OFF",
      },
    ]),
  });

  console.log(response);

  expect(2).toEqual(2);
});
