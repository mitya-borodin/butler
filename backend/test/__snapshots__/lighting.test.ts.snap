// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add lighting devices in group 1`] = `
Array [
  Object {
    "devices": Array [],
    "location": "КУХНЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ГОСТИНАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ИГРОВАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ВАННАЯ",
    "state": "OFF",
  },
]
`;

exports[`Add lighting devices in group 2`] = `
Array [
  Object {
    "brand": "FREON_2",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-2",
    "placeOfInstallation": "КУХНЯ",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
  Object {
    "brand": "FREON_1",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-1",
    "placeOfInstallation": "КУХНЯ",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
]
`;

exports[`Create lighting groups 1`] = `
Array [
  Object {
    "devices": Array [],
    "location": "КУХНЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ГОСТИНАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ИГРОВАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ВАННАЯ",
    "state": "OFF",
  },
]
`;

exports[`Get lighting group 1`] = `
Array [
  Object {
    "devices": Array [],
    "location": "КУХНЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ГОСТИНАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ИГРОВАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ВАННАЯ",
    "state": "OFF",
  },
]
`;

exports[`Get lighting groups 1`] = `
Array [
  Object {
    "devices": Array [],
    "location": "КУХНЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ГОСТИНАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ИГРОВАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ВАННАЯ",
    "state": "OFF",
  },
]
`;

exports[`Get lighting groups 2`] = `
Array [
  Object {
    "devices": Array [],
    "location": "ВАННАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ГОСТИНАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ИГРОВАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "КУХНЯ",
    "state": "OFF",
  },
]
`;

exports[`Move lighting devices to another group 1`] = `
Array [
  Object {
    "devices": Array [],
    "location": "КУХНЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ГОСТИНАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ИГРОВАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ВАННАЯ",
    "state": "OFF",
  },
]
`;

exports[`Move lighting devices to another group 2`] = `
Array [
  Object {
    "brand": "FREON_2",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-2",
    "placeOfInstallation": "КУХНЯ",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
  Object {
    "brand": "FREON_1",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-1",
    "placeOfInstallation": "КУХНЯ",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
]
`;

exports[`Move lighting devices to another group 3`] = `
Array [
  Object {
    "brand": "FREON_2",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
      Object {
        "placeOfInstallation": "NOT_INSTALLED",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
      Object {
        "placeOfInstallation": "ГОСТИНАЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-2",
    "placeOfInstallation": "ГОСТИНАЯ",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
  Object {
    "brand": "FREON_1",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
      Object {
        "placeOfInstallation": "NOT_INSTALLED",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
      Object {
        "placeOfInstallation": "ГОСТИНАЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-1",
    "placeOfInstallation": "ГОСТИНАЯ",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
]
`;

exports[`Remove lighting devices from group 1`] = `
Array [
  Object {
    "devices": Array [],
    "location": "КУХНЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ГОСТИНАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ИГРОВАЯ",
    "state": "OFF",
  },
  Object {
    "devices": Array [],
    "location": "ВАННАЯ",
    "state": "OFF",
  },
]
`;

exports[`Remove lighting devices from group 2`] = `
Array [
  Object {
    "brand": "FREON_2",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-2",
    "placeOfInstallation": "КУХНЯ",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
  Object {
    "brand": "FREON_1",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-1",
    "placeOfInstallation": "КУХНЯ",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
]
`;

exports[`Remove lighting devices from group 3`] = `
Array [
  Object {
    "brand": "FREON_2",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
      Object {
        "placeOfInstallation": "NOT_INSTALLED",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-2",
    "placeOfInstallation": "NOT_INSTALLED",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
  Object {
    "brand": "FREON_1",
    "currency": "RUB",
    "history": Array [
      Object {
        "placeOfInstallation": "КУХНЯ",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
      Object {
        "placeOfInstallation": "NOT_INSTALLED",
        "turnedOffAt": null,
        "turnedOnAt": null,
        "workedMs": null,
      },
    ],
    "images": Array [
      "image_1",
      "image_2",
      "image_3",
    ],
    "lightTemperatureKelvin": 4000,
    "lumens": "2700",
    "name": "lightning-device-1",
    "placeOfInstallation": "NOT_INSTALLED",
    "power": "24",
    "price": "600",
    "resourceMs": 30000,
    "sellersWebsite": "http://feron.ru",
    "state": "IN_STOCK",
  },
]
`;
